socks-proxy-agent.js

```diff
@@ -141,7 +141,12 @@ function connect (req, _opts, fn) {
     },
     command: 'connect'
   };
-
+  if (proxy.auth){
```

Comment:
I'd rather move this logic to the `SocksProxyAgent` constructor where the other `proxy` options are parsed.